core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -477,19 +468,26 @@ private boolean processInitial(TransferProcess process) {
         transferProcessStore.update(process);
         observable.invokeForEach(l -> l.provisioning(process));
 
-        provisionManager.provision(process).whenComplete((responses, throwable) -> {
-            if (throwable == null) {
-                onProvisionComplete(process.getId(), responses);
-            } else {
-                monitor.severe("Error during provisioning", throwable);
-                process.transitionError("Error during provisioning: " + throwable.getCause().getLocalizedMessage());
-                transferProcessStore.update(process);
-            }
-        });
+        provisionManager.provision(process)
+                .whenComplete((responses, throwable) -> {
+                    if (throwable == null) {
+                        onProvisionComplete(process.getId(), responses);
+                    } else {
+                        transitionToError(process.getId(), throwable.getCause(), "Error during provisioning");
+                    }
+                });
 
         return true;
     }
 
+    private void transitionToError(String id, Throwable throwable, String message) {
```

Comment:
Is reloading really absolutely necessary for `transitionError`? A transition to `ERROR` should be possible from every other state, so it shouldn't matter. I'd be ok with leaving it in for having a consistent programming model though.

Reply:
Probably not necessary, just for consistency. Also added a null check, probably not necessary but just for consistency too.